gramps_webapi/api/resources/reports.py

```diff
@@ -0,0 +1,214 @@
+"""Reports Plugin API resource."""
+
+import io
+import json
+import os
+import uuid
+from pathlib import Path
+from typing import BinaryIO, Dict
+
+from flask import Response, abort, send_file
+from gramps.cli.plug import CommandLineReport, cl_report
+from gramps.gen.db.base import DbReadBase
+from gramps.gen.filters import reload_custom_filters
+from gramps.gen.plug import CATEGORY_BOOK, CATEGORY_WEB, BasePluginManager
+from gramps.gen.utils.resourcepath import ResourcePath
+from webargs import fields, validate
+from webargs.flaskparser import use_args
+
+from ...const import REPORT_DEFAULTS, REPORT_MIMETYPES
+from ..util import get_dbstate
+from . import ProtectedResource
+from .emit import GrampsJSONEncoder
+
+_UNSUPPORTED = [CATEGORY_WEB, CATEGORY_BOOK]
+
+_EXTENSION_MAP = {"gvpdf": "pdf", "gspdf": "pdf"}
+
+
+def get_report_profile(
+    db_handle: DbReadBase, plugin_manager: BasePluginManager, report_data
+):
+    """Extract and return report attributes and options."""
+    module = plugin_manager.load_plugin(report_data)
+    option_class = getattr(module, report_data.optionclass)
+    report = CommandLineReport(
+        db_handle, report_data.name, report_data.category, option_class, {}
+    )
+    return {
+        "id": report_data.id,
+        "name": report_data.name,
+        "name_accell": report_data.name_accell,
+        "description": report_data.description,
+        "version": report_data.version,
+        "status": report_data.status,
+        "fname": report_data.fname,
+        "fpath": report_data.fpath,
+        "ptype": report_data.ptype,
+        "authors": report_data.authors,
+        "authors_email": report_data.authors_email,
+        "supported": report_data.supported,
+        "load_on_reg": report_data.load_on_reg,
+        "icons": report_data.icons,
+        "icondir": report_data.icondir,
+        "category": report_data.category,
+        "module": report_data.mod_name,
+        "reportclass": report_data.reportclass,
+        "require_active": report_data.require_active,
+        "report_modes": report_data.report_modes,
+        "option_class": report_data.optionclass,
+        "options_dict": report.options_dict,
+        "options_help": report.options_help,
+    }
```

Comment:
Stupid question, but this dict looks like it is always `"key": report_data.key`, couldn't that be abbreviated somehow? Perhaps using `report_data.__dict__`?

Reply:
Not stupid, I'll have to look at it again.  Possible and I just didn't think of it.

Reply:
Unfortunately we only want report specific attributes in the response here and that approach would include attributes associated with other plugin types.